fix(diagnostic): silence chistory command

yorickpeterse opened the pull request:
vim.diagnostic.set_list() uses chistory to restore the actively selected entry whenever necessary. This however also results in it displaying some output in the message bar, but this output isn't useful (and can even be distracting) when opening the quickfix window. This fixes this by silencing the chistory command.

yorickpeterse commented:
The commit refers to the output that can be seen in the bottom of this screenshot:
![Screenshot From 2024-12-23 04-10-27](https://github.com/user-attachments/assets/ff96d60e-b277-4a26-89d5-c653ef27cc15)

justinmk commented on runtime/lua/vim/diagnostic.lua:
```diff
@@ -874,7 +874,7 @@ local function set_list(loclist, opts)
       -- First navigate to the diagnostics quickfix list.
       local nr = vim.fn.getqflist({ id = qf_id, nr = 0 }).nr
-      api.nvim_command(nr .. 'chistory')
+      api.nvim_command('silent ' .. nr .. 'chistory')
```
```suggestion
      api.nvim_command('silent %dchistory')
```

yorickpeterse committed:
fix(diagnostic): silence chistory command

vim.diagnostic.set_list() uses chistory to restore the actively selected
entry whenever necessary. This however also results in it displaying
some output in the message bar, but this output isn't useful (and can
even be distracting) when opening the quickfix window. This fixes this
by silencing the chistory command.
